### ISSUE TITLE
What-If examples failing on Jupyter Notebooks

### ISSUE THREAD

**Issue by timoei** [tensorflow/tensorboard#2844]
Hi,

For some days I've tried to get What-If working on a Jupyter Notebook. I tried this both wirth Python 2.7 as well as Python 3.7. I followed the instructions provided in the ReadMe:

> pip install witwidget
jupyter nbextension install --py --symlink --sys-prefix witwidget
jupyter nbextension enable --py --sys-prefix witwidget

This is happening for the _SmileDetection_ and the _Income Classification_ example. When initiating WitWidget always the following error raises:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-0898599483f1> in <module>
     16     classifier, feature_spec).set_compare_estimator_and_feature_spec(
     17     classifier2, feature_spec).set_label_vocab(['Under 50K', 'Over 50K'])
---> 18 a = WitWidget(config_builder, height=tool_height_in_px)

/opt/conda/lib/python3.7/site-packages/witwidget/notebook/jupyter/wit.py in __init__(self, config_builder, height)
     61     """
     62     widgets.DOMWidget.__init__(self, layout=Layout(height='%ipx' % height))
---> 63     base.WitWidgetBase.__init__(self, config_builder)
     64     self.error_counter = 0
     65 

/opt/conda/lib/python3.7/site-packages/witwidget/notebook/base.py in __init__(self, config_builder)
     86       del copied_config['compare_adjust_attribution']
     87 
---> 88     self.config = copied_config
     89     self.set_examples(config['examples'])
     90     del copied_config['examples']

/opt/conda/lib/python3.7/site-packages/traitlets/traitlets.py in __set__(self, obj, value)
    583             raise TraitError('The "%s" trait is read-only.' % self.name)
    584         else:
--> 585             self.set(obj, value)
    586 
    587     def _validate(self, obj, value):

/opt/conda/lib/python3.7/site-packages/traitlets/traitlets.py in set(self, obj, value)
    572             # we explicitly compare silent to True just in case the equality
    573             # comparison above returns something other than True/False
--> 574             obj._notify_trait(self.name, old_value, new_value)
    575 
    576     def __set__(self, obj, value):

/opt/conda/lib/python3.7/site-packages/traitlets/traitlets.py in _notify_trait(self, name, old_value, new_value)
   1137             new=new_value,
   1138             owner=self,
-> 1139             type='change',
   1140         ))
   1141 

/opt/conda/lib/python3.7/site-packages/ipywidgets/widgets/widget.py in notify_change(self, change)
    603             if name in self.keys and self._should_send_property(name, getattr(self, name)):
    604                 # Send new state to front-end
--> 605                 self.send_state(key=name)
    606         super(Widget, self).notify_change(change)
    607 

/opt/conda/lib/python3.7/site-packages/ipywidgets/widgets/widget.py in send_state(self, key)
    487             state, buffer_paths, buffers = _remove_buffers(state)
    488             msg = {'method': 'update', 'state': state, 'buffer_paths': buffer_paths}
--> 489             self._send(msg, buffers=buffers)
    490 
    491 

/opt/conda/lib/python3.7/site-packages/ipywidgets/widgets/widget.py in _send(self, msg, buffers)
    735         """Sends a message to the model in the front-end."""
    736         if self.comm is not None and self.comm.kernel is not None:
--> 737             self.comm.send(data=msg, buffers=buffers)
    738 
    739     def _repr_keys(self):

/opt/conda/lib/python3.7/site-packages/ipykernel/comm/comm.py in send(self, data, metadata, buffers)
    119         """Send a message to the frontend-side version of this comm"""
    120         self._publish_msg('comm_msg',
--> 121             data=data, metadata=metadata, buffers=buffers,
    122         )
    123 

/opt/conda/lib/python3.7/site-packages/ipykernel/comm/comm.py in _publish_msg(self, msg_type, data, metadata, buffers, **keys)
     63         data = {} if data is None else data
     64         metadata = {} if metadata is None else metadata
---> 65         content = json_clean(dict(data=data, comm_id=self.comm_id, **keys))
     66         self.kernel.session.send(self.kernel.iopub_socket, msg_type,
     67             content,

/opt/conda/lib/python3.7/site-packages/ipykernel/jsonutil.py in json_clean(obj)
    189         out = {}
    190         for k,v in iteritems(obj):
--> 191             out[unicode_type(k)] = json_clean(v)
    192         return out
    193     if isinstance(obj, datetime):

/opt/conda/lib/python3.7/site-packages/ipykernel/jsonutil.py in json_clean(obj)
    189         out = {}
    190         for k,v in iteritems(obj):
--> 191             out[unicode_type(k)] = json_clean(v)
    192         return out
    193     if isinstance(obj, datetime):

/opt/conda/lib/python3.7/site-packages/ipykernel/jsonutil.py in json_clean(obj)
    189         out = {}
    190         for k,v in iteritems(obj):
--> 191             out[unicode_type(k)] = json_clean(v)
    192         return out
    193     if isinstance(obj, datetime):

/opt/conda/lib/python3.7/site-packages/ipykernel/jsonutil.py in json_clean(obj)
    189         out = {}
    190         for k,v in iteritems(obj):
--> 191             out[unicode_type(k)] = json_clean(v)
    192         return out
    193     if isinstance(obj, datetime):

/opt/conda/lib/python3.7/site-packages/ipykernel/jsonutil.py in json_clean(obj)
    175 
    176     if isinstance(obj, list):
--> 177         return [json_clean(x) for x in obj]
    178 
    179     if isinstance(obj, dict):

/opt/conda/lib/python3.7/site-packages/ipykernel/jsonutil.py in <listcomp>(.0)
    175 
    176     if isinstance(obj, list):
--> 177         return [json_clean(x) for x in obj]
    178 
    179     if isinstance(obj, dict):

/opt/conda/lib/python3.7/site-packages/ipykernel/jsonutil.py in json_clean(obj)
    195 
    196     # we don't understand it, it's probably an unserializable object
--> 197     raise ValueError("Can't clean for JSON: %r" % obj)

ValueError: Can't clean for JSON: features {
  feature {
    key: "Age"
    value {
      int64_list {
        value: 25
      }
    }
  }
```
By the way the code provided in the Colab notebooks isn't working with Tensorflow 2.0. I installed and used version 1.15 for that reason.

Best regards,
Timo

**Comment by timoei** [comment]
Because What-If worked on another machine before, I tried the version which installed there:

> pip install witwidget==**1.2**
jupyter nbextension install --py --symlink --sys-prefix witwidget
jupyter nbextension enable --py --sys-prefix witwidget

This is still working. I hope this maybe helps to identify the issue.

Best,
Timo

**Comment by jameswex** [comment]
Thanks for pointing this out. It seems we had a regression for Jupyter notebooks in certain cases in witwidget 1.4.2 and will fix this and release a new patch version ASAP.

**Comment by jameswex** [comment]
We have released witwidget 1.4.5 which fixes this issue. Sorry for the trouble. Please reopen if you continue to have trouble after updating to the newest version.
